win32/src/threadpool_win32.c

```diff
@@ -661,13 +671,17 @@ void threadpool_timer_destroy(TIMER_INSTANCE_HANDLE timer)
     }
     else
     {
+        /* Codes_SRS_THREADPOOL_WIN32_07_002: [ threadpool_timer_destroy shall indicate the timer is in destroying state. ]*/
+        (void)interlocked_exchange(&timer->is_destroying, 1);
         /* Codes_SRS_THREADPOOL_WIN32_42_012: [ threadpool_timer_destroy shall call SetThreadpoolTimer with NULL for pftDueTime and 0 for msPeriod and msWindowLength to cancel ongoing timers. ]*/
         /* Codes_SRS_THREADPOOL_WIN32_42_013: [ threadpool_timer_destroy shall call WaitForThreadpoolTimerCallbacks. ]*/
         threadpool_internal_cancel_timer_and_wait(timer->timer);
 
         /* Codes_SRS_THREADPOOL_WIN32_42_014: [ threadpool_timer_destroy shall call CloseThreadpoolTimer. ]*/
         CloseThreadpoolTimer(timer->timer);
 
+        /* Codes_SRS_THREADPOOL_WIN32_07_003: [ threadpool_timer_destroy shall indicate the timer is destroyed. ]*/
+        (void)interlocked_exchange(&timer->is_destroying, 0);

```

Comment:
why are we setting it to 0? we are destroying it...

Reply:
will move it after the free